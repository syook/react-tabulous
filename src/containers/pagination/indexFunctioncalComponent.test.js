import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import PaginationProvider from './indexFunctioncalComponent';

configure({ adapter: new Adapter() });

const data = [
  { Description: 'Dev', Name: 'David', id: 1, objIndex: 0 },
  { Description: 'QA', Name: 'John', id: 2, objIndex: 1 },
];

describe('SortProvider', () => {
  it('should render without crashing', () => {
    shallow(<PaginationProvider data={data} />);
  });

  it('should match snapshot if data is not empty', () => {
    const tree = shallow(<PaginationProvider data={data} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if data is empty', () => {
    const tree = shallow(<PaginationProvider data={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should contain Table component', () => {
    const wrapper = shallow(<PaginationProvider data={data} />);
    expect(wrapper.find('Table').exists()).toBe(true);
  });

  it('Table should contain sortable as true', () => {
    const wrapper = shallow(<PaginationProvider data={data} />);
    expect(wrapper.find('Table').props().sortable).toBe(true);
  });

  it('Table should contain celled as true', () => {
    const wrapper = shallow(<PaginationProvider data={data} />);
    expect(wrapper.find('Table').props().celled).toBe(true);
  });

  it('Table should contain padded as true', () => {
    const wrapper = shallow(<PaginationProvider data={data} />);
    expect(wrapper.find('Table').props().padded).toBe(true);
  });

  it('should contain Pagination component', () => {
    const wrapper = shallow(<PaginationProvider data={data} />);
    expect(wrapper.find('Pagination').exists()).toBe(true);
  });

  it('Pagination should contain rowCount as passed prop count', () => {
    const wrapper = shallow(<PaginationProvider data={data} count={2} />);
    expect(wrapper.find('Pagination').props().rowCount).toBe(2);
  });
});
