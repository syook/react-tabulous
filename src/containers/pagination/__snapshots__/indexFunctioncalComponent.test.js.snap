// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortProvider should match snapshot if data is empty 1`] = `
<Fragment>
  <div
    className="scrollable-table tableFixHead null"
    style={
      Object {
        "marginTop": "10px",
        "maxWidth": "100%",
      }
    }
  >
    <Table
      as="table"
      celled={true}
      className="tableStyle left aligned table-fixed"
      padded={true}
      sortable={true}
    >
      <ContextProvider
        value={
          Object {
            "currentPage": 1,
            "data": Array [],
            "numberOfColumns": 30,
            "numberOfPages": 0,
            "rawData": undefined,
            "resetToFirstPage": [Function],
            "rowCount": 0,
            "rowsPerPage": Object {
              "label": "10 Items",
              "value": 10,
            },
            "startIndex": 0,
          }
        }
      />
    </Table>
  </div>
  <Pagination
    currentPage={1}
    data={Array []}
    handleDirectionClick={[Function]}
    handlePageClick={[Function]}
    numberOfColumns={30}
    numberOfPages={0}
    onSelectRowsPerPage={[Function]}
    pageRange={Array []}
    rowCount={0}
    rowsPerPage={
      Object {
        "label": "10 Items",
        "value": 10,
      }
    }
    setCurrentPage={[Function]}
  />
</Fragment>
`;

exports[`SortProvider should match snapshot if data is not empty 1`] = `
<Fragment>
  <div
    className="scrollable-table tableFixHead null"
    style={
      Object {
        "marginTop": "10px",
        "maxWidth": "100%",
      }
    }
  >
    <Table
      as="table"
      celled={true}
      className="tableStyle left aligned table-fixed"
      padded={true}
      sortable={true}
    >
      <ContextProvider
        value={
          Object {
            "currentPage": 1,
            "data": Array [
              Object {
                "Description": "Dev",
                "Name": "David",
                "id": 1,
                "objIndex": 0,
              },
              Object {
                "Description": "QA",
                "Name": "John",
                "id": 2,
                "objIndex": 1,
              },
            ],
            "numberOfColumns": 30,
            "numberOfPages": 1,
            "rawData": undefined,
            "resetToFirstPage": [Function],
            "rowCount": 2,
            "rowsPerPage": Object {
              "label": "10 Items",
              "value": 10,
            },
            "startIndex": 0,
          }
        }
      />
    </Table>
  </div>
  <Pagination
    currentPage={1}
    data={
      Array [
        Object {
          "Description": "Dev",
          "Name": "David",
          "id": 1,
          "objIndex": 0,
        },
        Object {
          "Description": "QA",
          "Name": "John",
          "id": 2,
          "objIndex": 1,
        },
      ]
    }
    handleDirectionClick={[Function]}
    handlePageClick={[Function]}
    numberOfColumns={30}
    numberOfPages={1}
    onSelectRowsPerPage={[Function]}
    pageRange={
      Array [
        1,
      ]
    }
    rowCount={2}
    rowsPerPage={
      Object {
        "label": "10 Items",
        "value": 10,
      }
    }
    setCurrentPage={[Function]}
  />
</Fragment>
`;
